scripts/demangle.py

```diff
@@ -59,46 +82,110 @@ def demangle(func: str):
-            except:
-                print('ERROR: ',json_file_name, "Json file illegal")
+                data = json.load(f)
+            except json.JSONDecodeError:
+                warning(f"ERROR: {json_file_name}  Json file illegal")
                 continue
-    for file_func_name in data:
-        # if split not matched, need to add a check
-        mangle_func_name = file_func_name.split(file_name)
-        if len(mangle_func_name)<2:
-            raise ValueError(f"ERROR: File '{file_name}' not found, may open wrong json file")
-        else:
-            mangle_func_name = mangle_func_name[1]
-            demangle_func_name = demangle(mangle_func_name)
-            # Add a check demangle is correct?
-            if mangle_func_name == demangle_func_name or demangle_func_name=='':
-                print('ERROR: '+mangle_func_name+' demangle incorrect or unable to demangle')
+        info(f"NUMBER OF FUNCTION: {len(data)}")
+        pre_file_path, code_content = "", ""
+        for cnt in range(len(data)):
+            file_func_name = ""
+            for file_func_name_ in data[cnt]:
+                file_func_name = file_func_name_
+
+            # if split not matched, need to add a check
+            splited_file_func_name = file_func_name.split("?")
+            file_path, mangle_func_name = (
+                splited_file_func_name[0],
+                splited_file_func_name[1],
+            )
+            # if this is c file then the mangle does not exist
+            try:
+                demangle_func_name = cpp_demangle.demangle(mangle_func_name)
+            except ValueError:
+                demangle_func_name = mangle_func_name
+
+            if demangle_func_name == "":  # mangle_func_name == demangle_func_name
+                warning(
+                    f"ERROR: {mangle_func_name} demangle incorrect or unable to demangle"
+                )
+                data[cnt][file_func_name] = (
+                    "ERROR: "
+                    + mangle_func_name
+                    + " demangle incorrect or unable to demangle"
+                )
             else:
-                func_name = demangle_func_name.split('(')[0]
-                # print('looking for func ', func_name, ' in  file ', file_path)
-                func_content = extract_func_code(func_name,file_path)
-                print(func_content)
+                func_name = demangle_func_name.split("(")[0]
+                # Try to get code_content, the json file may provide the path or not
+                # only get code when pre_file_path!=file_path
+                if pre_file_path != file_path:
+                    code_content = get_source_code(file_path, file_name, proj_name)
+                    pre_file_path = file_path
+                    info(f"Open source file: {pre_file_path}")
+                    with open(
+                        "./output/" + proj_name + "/" + json_file_name + ".txt", "w"
```

Comment:
use abs path `os.path.join(OSSFUZZ_SCRIPTS_HOME, "output", proj_name ...)` might be better